### PR TITLE
[local-authentication] add doc comments in source

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-1617

To utilize the auto generation feature for API section on the docs website the package source code must contain the relevant documentation.

# How

This PR moves the current documentation of `LocalAuthentication` API - https://docs.expo.io/versions/latest/sdk/local-authentication - with minor tweaks and additions to the package source code.

To full support the migration to automated docs extraction I had to introduce few improvements and fixes to the code:
* display enum values (if enum has values set) and order those enums by value not name
* correctly render type if it's directly set to one of `boolean` values

This PR also includes the small fix for the deprecation note rendering issue reported in #13729.

I have also generalized the `LocalAuthenticationResult` type, but I'm not sure if such change should be reported in the package Changelog. If it should, let me know, I will add an entry.

# Test Plan

The changes and generated content have been tests running docs website on `localhost`.

# Preview

### Page + sidebar
Screenshot 2021-07-26 at 11 27 15

### Enums
Screenshot 2021-07-26 at 11 26 54

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [X] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [X] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/components/plugins/api/APISectionEnums.tsx [hunk 1]
```diff
@@ -10,18 +10,37 @@ export type APISectionEnumsProps = {
...
 };
 
+const sortByValue = (a: EnumValueData, b: EnumValueData) => {
+  if (a.defaultValue && b.defaultValue) {
+    if (a.defaultValue.includes(`'`) && b.defaultValue.includes(`'`)) {
+      return a.defaultValue.localeCompare(b.defaultValue);
+    } else {
+      const vA = parseInt(a.defaultValue, 10);
+      const vB = parseInt(b.defaultValue, 10);
+      return vA > vB ? 1 : -1;
```
Comment (bbarthec) [inline docs/components/plugins/api/APISectionEnums.tsx:20]: I saw a lot that when you compare two numbers you simply subtract one from another obtaining as a result `negative number`, `0` or `positive number` 🧙 
```suggestion
      return vA - vB;
```

File: docs/components/plugins/api/APISectionUtils.tsx [hunk 2]
```diff
@@ -195,6 +196,8 @@ export const resolveTypeName = ({
     );
   } else if (type === 'query' && queryType) {
     return queryType.name;
+  } else if (type === 'literal' && (value === true || value === false)) {
```
Comment (bbarthec) [inline docs/components/plugins/api/APISectionUtils.tsx:199]:
```suggestion
  } else if (type === 'literal' && typeof value === 'boolean') {
```

**Review by bbarthec — APPROVED** [review]
Looks good to me 👍

File: docs/components/plugins/api/APISectionEnums.tsx (hunk 1 above)
Comment (Simek) [reply docs/components/plugins/api/APISectionEnums.tsx:20]: Interesting, I will check this locally before the merge, thank you! :)

**Pull request merged**